File: Programmers/Lv2/괄호 회전하기/Ohjemin2.py
from collections import deque
def solution(s):
    answer = 0
    queue = deque()
    for i in s:
        queue.append(i)
    for i in range(len(s)):            
        if rotate(''.join(list(queue))) == 0:
            answer +=1
        queue.rotate(-1)
    return answer

def rotate(s):
    while True:
        if "()" in s:
            s = s.replace("()","")
        elif "[]" in s:
            s = s.replace("[]","")
        elif "{}" in s:
            s = s.replace("{}","")
        else: 
            return -1 if s else 0  

File: Programmers/Lv2/괄호 회전하기/Ohjemin2.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        Deque<String> q = new ArrayDeque<>();
        String[] arr = s.split("");
        for(String i : arr) q.add(i);
        for(int i=0;i<s.length();i++){                      
            if(rotate(q)) answer++;
            q.add(q.poll());
        }
        return answer;
    }
    public static boolean rotate(Deque<String> q){
        String tmp = String.join("",q);  
        while(true){
            if(tmp.contains("()")) tmp = tmp.replace("()","");
            else if(tmp.contains("[]")) tmp = tmp.replace("[]","");
            else if(tmp.contains("{}")) tmp = tmp.replace("{}","");
            else return tmp.length() == 0 ? true : false;
        }
    }
}
